UnityProject/Assets/Scripts/Objects/Other/SecBarrier.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Objects.Security
+{
+	[RequireComponent(typeof(AccessRestrictions))]
+	[RequireComponent(typeof(Construction.WrenchSecurable))]
+	public class SecBarrier : MonoBehaviour, IPredictedCheckedInteractable<HandApply>
+	{
+		private AccessRestrictions accessRestrictions;
+		public AccessRestrictions AccessRestrictions
+		{
+			get
+			{
+				if (accessRestrictions == false)
+				{
+					accessRestrictions = GetComponent<AccessRestrictions>();
+				}
+				return accessRestrictions;
+			}
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			//start with the default HandApply WillInteract logic.
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+			//Only go through with interaction if correct access restriction
+			if (AccessRestrictions.CheckAccess(interaction.Performer))
+			{
+				//Player needs wrench to interact
+				if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench))
```

Comment:
Uhm wouldn't this mean that if you have the acces but not the tool you suddenly get a true? Maybe it is better to join both conditions with an ``&&`` and return the opposite of it.

``return !(AccessRestrictions.CheckAccess(interaction.Performer) && (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench))``

Reply:
Ah yes, thanks and resolved!